sync/syncer.go

```diff
@@ -226,49 +223,47 @@ func (s *Syncer) getLayerBlockIDs(index types.LayerID) (chan types.BlockID, erro
 }
 
 func (s *Syncer) getIdsForHash(m map[string]p2p.Peer, index types.LayerID) (chan types.BlockID, error) {
-	reqCounter := 0
 
 	ch := make(chan []types.BlockID, len(m))
-	wg := sync.WaitGroup{}
-	for _, v := range m {
-		c, err := s.sendLayerBlockIDsRequest(v, index)
+	kill := make(chan struct{})
+	for _, peer := range m {
+		c, err := s.sendLayerBlockIDsRequest(peer, index)
 		if err != nil {
-			s.Error("could not fetch layer ", index, " block ids from peer ", v) // todo recover from this
+			s.Error("could not send layer %v ids ", index, " from peer ", peer) // todo recover from this
 			continue
 		}
-		reqCounter++
-		wg.Add(1)
-		go func() {
-			if v := <-c; v != nil {
-				ch <- v
+		go func(p p2p.Peer) {
+			select {
+			case v := <-c:
+				if v != nil {
+					ch <- v
+				}
+			case <-kill:
+				s.Info("ids request to %v timed out", p)
```

Comment:
maybe cloes ch here (not a must)

Reply:
its closed by the routine that closed the kill chan